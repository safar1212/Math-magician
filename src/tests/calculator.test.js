import Calculator from '../components/calculator';
import  {render} from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Calculator page of the App', () => {
    test('Builds the snapchot of the Calculator component safely', () => {
      const calc = renderer.create(<Calculator />);
      expect(calc.toJSON()).toMatchSnapshot();
    });

    test('Render Calculator component', () => {
      render(<Calculator />);
    });

  });






// describe('Calculator', () => {
//   test('renders calculator component', () => {
//     render(<Calculator />);

//     screen.getByRole(div);
//   });
// });
