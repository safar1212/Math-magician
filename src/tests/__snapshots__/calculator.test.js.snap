// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page of the App Builds the snapchot of the Calculator component safely 1`] = `
<div
  className="calculator-div"
>
  <div
    className="input"
  />
  <div
    className="buttons-div"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
